src/main/java/bio/terra/service/dataset/DatasetService.java

```diff
@@ -265,6 +265,22 @@ public String deleteTabularData(
     }
   }
 
+  public void lockDataset(UUID datasetId, String flightId, boolean sharedLock) {
```

Comment:
IMO it's redundant to include `dataset` in this method name, since it's in `DatasetService`. Are there other kinds of locking that we might expect this class to handle?